amqp-impl/src/main/java/io/streamnative/pulsar/handlers/amqp/ExchangeContainer.java

```diff
@@ -57,18 +57,68 @@ public static void putExchange(NamespaceName namespaceName, String exchangeName,
         });
     }
 
-    public static AmqpExchange getExchange(NamespaceName namespaceName, String exchangeName) {
-        if (namespaceName == null || StringUtils.isEmpty(exchangeName)) {
-            return null;
+    public CompletableFuture<AmqpExchange> asyncGetExchange(PulsarService pulsarService,
+                                                            NamespaceName namespaceName,
+                                                            String exchangeName,
+                                                            boolean createIfMissing,
+                                                            String exchangeType) {
+        CompletableFuture<AmqpExchange> amqpExchangeCompletableFuture = new CompletableFuture<>();
+        if (StringUtils.isEmpty(exchangeType) && createIfMissing) {
+            log.error("exchangeType should be set when createIfMissing is true");
+            amqpExchangeCompletableFuture.complete(null);
         }
-        Map<String, AmqpExchange> map = exchangeMap.getOrDefault(namespaceName, null);
-        if (map == null) {
-            return null;
-        }
-        return map.getOrDefault(exchangeName, null);
+        executor.execute(() -> {
+            if (namespaceName == null || StringUtils.isEmpty(exchangeName)) {
+                amqpExchangeCompletableFuture.complete(null);
+            }
+            Map<String, AmqpExchange> map = exchangeMap.getOrDefault(namespaceName, null);
+            if (map == null || map.getOrDefault(exchangeName, null) == null) {
+                // check pulsar topic
+                if (pulsarService.getState() != PulsarService.State.Started) {
+                    amqpExchangeCompletableFuture.completeExceptionally(
+                            new PulsarServerException("PulsarService not start"));
+                }
+                String topicName = PersistentExchange.getExchangeTopicName(namespaceName, exchangeName);
+                CompletableFuture<Topic> topicCompletableFuture =
+                        amqpTopicManager.getTopic(topicName, createIfMissing);
+                topicCompletableFuture.whenComplete((topic, throwable) -> {
+                    if (throwable != null) {
+                        log.error("Get topic error:{}", throwable.getMessage());
+                        amqpExchangeCompletableFuture.complete(null);
+                    } else {
+                        if (null == topic) {
+                            log.error("Exchange topic not existed, exchangeName:{}", exchangeName);
```

Comment:
NIT: Consider using this wording instead: "The exchange topic did not exist.  namespace{}, exchangeName: {}"